app/gateway/database/postgres/account/create_test.go

```diff
@@ -0,0 +1,104 @@
+package account
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"stonehenge/app/core/entities/account"
+	"stonehenge/app/core/types/password"
+	"stonehenge/app/gateway/database/postgres/postgrestest"
+	"stonehenge/app/gateway/database/postgres/transaction"
+	"testing"
+)
+
+func TestCreate(t *testing.T) {
+	db, err := postgrestest.NewCleanDatabase()
+	if err != nil {
+		t.Fatalf("could not get database: %v", err)
+	}
+
+	tx := transaction.NewTransaction(db)
+
+	type args struct {
+		ctx     context.Context
+		account account.Account
+	}
+
+	type test struct {
+		name    string
+		args    args
+		before  func(test, account.Repository) error
+		wantErr error
+	}
+
+	tests := []test{
+
+		// Should return Account with generated fields
+		{
+			name: "return Account with generated fields",
+			args: args{ctx: context.Background(), account: account.Account{
+				Document: "05161964057",
+				Secret:   password.From("12345678"),
+				Name:     "Spencer Reis",
+				Balance:  5000,
+			}},
+		},
+
+		// Should return ErrAlreadyExist for duplicate account
+		{
+			name: "return ErrAlreadyExist for duplicate account",
+			before: func(test test, repo account.Repository) error {
+				_, err := postgrestest.PopulateAccounts(test.args.ctx, account.Account{
+					Document: "05161964057",
+					Secret:   password.From("12345678"),
+					Name:     "Spencer Reis",
+					Balance:  5000,
+				})
+				return err
+			},
+			args: args{ctx: context.Background(), account: account.Account{
+				Document: "05161964057",
+				Secret:   password.From("12345678"),
+				Name:     "Spencer Reis",
+				Balance:  5000,
+			}},
+			wantErr: account.ErrAlreadyExist,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			defer postgrestest.RecycleDatabase(test.args.ctx)
+
+			repo := NewRepository(db)
+
+			if test.before != nil {
+				err := test.before(test, repo)
+				if err != nil {
+					t.Fatalf("error running routine before: %v", err)
+				}
+			}
+			ctx, err := tx.Begin(test.args.ctx)
+			if err != nil {
+				t.Error(err)
+			}
+
+			acc, err := repo.Create(ctx, test.args.account)
+			if err != nil {
+				tx.Rollback(ctx)
+				assert.ErrorIs(t, test.wantErr, err)
+				return
+			}
+
+			if err := tx.Commit(ctx); err != nil {
+				tx.Rollback(ctx)
+				assert.ErrorIs(t, test.wantErr, err)
+			}
```

Comment:
você pode chamar um defer tx.Rollback(ctx) logo abaixo do begin para evitar chamá-lo em todos os casos de erro.

```suggestion
                        defer tx.Rollback(ctx)
			acc, err := repo.Create(ctx, test.args.account)
			if err != nil {
				assert.ErrorIs(t, test.wantErr, err)
				return
			}

			if err := tx.Commit(ctx); err != nil {
				assert.ErrorIs(t, test.wantErr, err)
			}
```

Reply:
Boa, valeu!